presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationFromAnnotationsParser.java

```diff
@@ -197,6 +200,23 @@ public static Method getCombineFunction(Class<?> clazz, Class<?> stateClass)
         return inputFunctions;
     }
 
+    private static Optional<Method> getRemoveInputFunction(Class<?> clazz, Class<?> stateClass, Method inputFunction)
+    {
+        // Only include methods which take the same parameters as the corresponding input function
+        List<Method> removeInputFunctions = FunctionsParserHelper.findPublicStaticMethodsWithAnnotation(clazz, RemoveInputFunction.class).stream()
+                .filter(method ->
+                        Arrays.equals(method.getParameterTypes(), inputFunction.getParameterTypes())
+                                && Arrays.deepEquals(method.getParameterAnnotations(), inputFunction.getParameterAnnotations()))
+                .collect(toImmutableList());
+
+        // Note that a missing removeInput function is not an error -- they are optional.
+        checkArgument(removeInputFunctions.size() <= 1, "Ambiguous aggregation removeInput functions");
```

Comment:
I think it's rather `checkState`, not an `checkArgument` because it may happen because of the wrong content of the class not because someone passed wrong argument to the method, although I understand the transitivity of "wrong method comes from the wrong class which was passed as an argument"...